File: automobileinsuranceapi/src/main/java/com/hexaware/automobileinsurance/service/PolicyService.java
package com.hexaware.automobileinsurance.service;

import java.util.List;

import com.hexaware.automobileinsurance.dto.PolicyDTO;
import com.hexaware.automobileinsurance.model.Policy;

public interface PolicyService {

    Policy addPolicy(PolicyDTO dto);

    Policy getPolicyById(Integer policyId);

    List<Policy> getAllPolicies();

    Policy updatePolicy(Integer policyId, PolicyDTO dto);

    void deletePolicy(Integer policyId);
}

File: automobile-insurance-frontend/src/services/PolicyService.js
import api from "../api/api";

const PolicyService = {

    getAllPolicies() {
        return api.get("/policies");
    },

    getPolicyById(id) {
        return api.get(`/policies/${id}`);
    },

    addPolicy(policy) {
        return api.post("/policies", policy);
    },

    updatePolicy(id, policy) {
        return api.put(`/policies/${id}`, policy);
    },

    deletePolicy(id) {
        return api.delete(`/policies/${id}`);
    }

};

export default PolicyService;